File: Ontology/HasKey.cs
using System.Collections.Generic;
using System.Linq;

namespace Ontology
{
    public class HasKey:
        Axiom,
        IHasKey
    {
        private readonly IClassExpression               _classExpression;
        private readonly IList<IDataPropertyExpression> _dataPropertyExpressions;

        public HasKey(
            IOntology                        ontology,
            IClassExpression                 classExpression,
            params IDataPropertyExpression[] dataPropertyExpressions
            ) : base(ontology)
        {
            _classExpression         = classExpression;
            _dataPropertyExpressions = dataPropertyExpressions;
        }

        IClassExpression IHasKey.ClassExpression => _classExpression;

        IList<IDataPropertyExpression> IHasKey.DataPropertyExpressions => _dataPropertyExpressions;
    }
}

File: Web/ClientApp/src/app/Ontology/HasKey.ts
import { Axiom } from "./Axiom";
import { IClassExpression } from "./IClassExpression";
import { IHasKey } from "./IHasKey";
import { IOntology } from "./IOntology";
import { IDataPropertyExpression } from "./IPropertyExpression";

export class HasKey
    extends Axiom
    implements IHasKey
{
    constructor(
        ontology                      : IOntology,
        public ClassExpression        : IClassExpression,
        public DataPropertyExpressions: IDataPropertyExpression[]
        )
    {
        super(ontology);
    }
}
